### ISSUE TITLE
Missing check for whether device is None in data.Field.numericalize

### ISSUE THREAD

**Issue by dogancan** [pytorch/text#27]
I'm getting the following error when calling [`data.Field.numericalize`](https://github.com/pytorch/text/blob/master/torchtext/data.py#L199) without a device argument. It seems like there needs to be a check for whether `device is None` before entering [this context manager](https://github.com/pytorch/text/blob/master/torchtext/data.py#L233).

```
Traceback (most recent call last):
  File "./train.py", line 102, in <module>
    main()
  File "./train.py", line 83, in main
    for step, batch in enumerate(train_iter):
  File "build/bdist.linux-x86_64/egg/torchtext/data.py", line 579, in __iter__
  File "build/bdist.linux-x86_64/egg/torchtext/data.py", line 482, in __init__
  File "build/bdist.linux-x86_64/egg/torchtext/data.py", line 233, in numericalize
  File "/home/dogan/anaconda2/lib/python2.7/site-packages/torch/cuda/__init__.py", line 132, in __enter__
    torch._C._cuda_setDevice(self.idx)
RuntimeError: invalid argument to setDevice
```

**Comment by jekbradbury** [comment]
Yeah, that's right. I think `device=None` used to be the same as no context manager (or maybe that was just in Chainer) but it's definitely not true now.

**Comment by jekbradbury** [comment]
Fixed in 15bf940
